fuse_core/include/fuse_core/throttled_callback.h

```diff
@@ -60,11 +60,12 @@ class ThrottledCallback
    * @param[in] keep_callback   The callback to call when kept, i.e. not dropped. Defaults to nullptr
    * @param[in] drop_callback   The callback to call when dropped because of the throttling. Defaults to nullptr
    * @param[in] throttle_period The throttling period duration in seconds. Defaults to 0.0, i.e. no throttling
-   * @param[in] use_wall_time   Whether to use ros::WallTime or not. Defaults to false
+   * @param[in] use_wall_time   Whether to use wall time or not. Defaults to false
    */
   ThrottledCallback(Callback&& keep_callback = nullptr,  // NOLINT(whitespace/operators)
                     Callback&& drop_callback = nullptr,  // NOLINT(whitespace/operators)
-                    const ros::Duration& throttle_period = ros::Duration(0.0), const bool use_wall_time = false)
+                    const rclcpp::Duration& throttle_period = rclcpp::Duration(0,0),
+                    const bool use_wall_time = false)
```

Comment:
I'd recommend requiring the clock to be passed in instead of a `use_wall_time` flag.

Reply:
This so far has been deferred: https://github.com/locusrobotics/fuse/pull/283/files#diff-250689f3de589b83ad938a77e9a67784dd33d4321363870ba7c1cf742182c44c

The way it's implemented currently is that there's a boolean ROS parameter ([throttle_use_wall_time](https://github.com/locusrobotics/fuse/blob/devel/fuse_models/include/fuse_models/parameters/acceleration_2d_params.h#L75)) though (it then eventually gets passed as the flag).. If we are to pass in a clock type, does that parameter have to change as well? If so what parameter should it become? (String? Int?)

I'm inclined to keep the flag and leave the boolean ROS parameter alone so there's parameter parity between the ROS 1 and ported version of Fuse.

Reply:
> I'm inclined to keep the flag and leave the boolean ROS parameter alone so there's parameter parity between the ROS 1 and ported version of Fuse.

I agree about keeping the parameter `throttle_use_wall_time` the same.

A clock will need to be passed in to use anything other than wall time because a ROS clock needs to be [attached to a time source](https://github.com/ros2/rclcpp/blob/b9b1468d15c7ddc697c079e6934d54f183294280/rclcpp/src/rclcpp/time_source.cpp#L92).  It can't be created inside of `ThrottledCallback()`. I'm not sure how to plumb that [here](https://github.com/locusrobotics/fuse/blob/07676a8a47ad00b605153d6d98aa58dce663a39e/fuse_models/src/twist_2d.cpp#L55) though. Maybe `fuse_core::AsyncSensorModel` can have a way for subclasses to get a ROS clock off of it?

Reply:
That's a good point, I'll be adding node interfaces and node interface getters to the classes that are currently storing node handles. I'll probably use them to obtain the NodeClockInterface and then get the clock that way :thinking: 

I'll defer this and add a TODO pending that change (which should be the next thing to change for fuse_core anyway!)